### PR TITLE
#164823820 (v2) Paginate room events

### PR DESCRIPTION
**Pull request by verenceLola**
### What does this PR do?
Paginate analytics for daily room events
### Description of the task to be completed?
Currently, when querying the analytics for daily room events, the result isn't paginated. This PR adds functionality to paginate the results when both the `page` number and items `perPage` are provided.
### How should this be manually tested?
 - Clone the branch and follow the following steps to [setup](https://github.com/andela/mrm_api).
 - Checkout by running `git checkout ft-paginate-room-events-164823820`
 - Perform the query below. This will return all the rooms without pagination 
```
query{
    analyticsForDailyRoomEvents(startDate:"jan 10 2019", endDate:"feb 10 2019"){
        DailyRoomEvents {
            day
            events {
                noOfParticipants,
                eventSummary,
                startTime,
                endTime,
                roomName,
                eventId
            }
        }
    }
}
```
 - query analytics for daily room events with pagination.
```
query{
    analyticsForDailyRoomEvents(startDate:"jan 10 2019",page:3, perPage:4, endDate:"feb 10 2019"){
        DailyRoomEvents {
            day
            events {
                noOfParticipants,
                eventSummary,
                startTime,
                endTime,
                roomName,
                eventId
            }
        }
    hasNext,
    hasPrevious,
    pages
    }
}
```
### Any background context you want to provide?
N/A
### What are the relevant pivotal tracker stories?
[#164823820](https://www.pivotaltracker.com/story/show/164823820)
### Screenshots
 - #### Query analytics for daily room events without pagination.
image

 - #### Query analytics for daily rooms events with pagination.
image

### CODE REVIEW

File: api/room/schema_query.py [hunk 1]
```diff
@@ -324,8 +333,21 @@ def resolve_analytics_for_daily_room_events(self, info,
                     events=daily_events
                 )
             )
-
-        return all_days_events
+        if page and per_page:
+            paginated_results = ListPaginate(iterable=all_days_events, per_page=per_page, page=page)  # noqa: E501
```
Comment (nakatuddesuzan) [inline api/room/schema_query.py]: Hello @verenceLola , Is it possible to get rid of this noqa?

Comment (verenceLola) [reply api/room/schema_query.py]: This has been worked on. Thanks for the observation @nakatuddesuzan

File: fixtures/room/daily_room_events_fixture.py [hunk 2]
```diff
@@ -40,8 +47,54 @@
...
                 }
             ]
-        }]
+        }
+      ]
+    }
+  }
+}
+
+many_daily_room_events_query = '''
```
Comment (nakatuddesuzan) [inline fixtures/room/daily_room_events_fixture.py]: This query name `many_daily_room_events_query` is a bit confusing. Please use one that is more descriptive.

File: tests/test_rooms/test_daily_room_events.py [hunk 3]
```diff
@@ -21,3 +23,10 @@ def test_analytics_for_daily_events_wrong_format_date(self):
             daily_room_events_wrong_date_format_query,
             daily_events_wrong_date_format_response
         )
+
+    def test_analytics_for_daily_events_pagination(self):
```
Comment (ThaDeveloper) [inline tests/test_rooms/test_daily_room_events.py]: @verenceLola Please provide docstrings for this method.

File: fixtures/room/daily_room_events_fixture.py (hunk 2 above)
Comment (verenceLola) [reply fixtures/room/daily_room_events_fixture.py]: @nakatuddesuzan This has been worked on. Thanks for the feedback

File: api/room/schema_query.py [hunk 4]
```diff
@@ -53,6 +53,13 @@ class DailyEvents(graphene.ObjectType):
     events = graphene.List(CalendarEvent)
 
 
+class PaginatedEvents(graphene.ObjectType):
```
Comment (RachaelNantale) [inline api/room/schema_query.py]: Great work @verenceLola, Please provide the documentation for this class.

Comment (verenceLola) [reply api/room/schema_query.py]: This has been worked on @RachaelNantale .

File: tests/test_rooms/test_daily_room_events.py (hunk 3 above)
Comment (verenceLola) [reply tests/test_rooms/test_daily_room_events.py]: This has been resolved. Thanks for the feedback @ThaDeveloper

**Comment by verenceLola** [comment]
> Update your branch with the base branch

This has been resolved.

File: api/room/schema_query.py (hunk 4 above)
Comment (dnuwa) [reply api/room/schema_query.py]: good work @verenceLola, you could also go ahead and add some field descriptions. something like 
`pages = graphene.Int(description="The pages number field of the events")`

**Comment by verenceLola** [comment]
> Handle a case when the page number, as well as the number of items per oage provided, does not have any results. Write the test for this too

Thanks for the feedback @Bonifase .This has been worked on.

**Pull request merged**